system-parachains/asset-hubs/asset-hub-polkadot/tests/tests.rs

```diff
@@ -657,3 +670,600 @@ asset_test_utils::include_create_and_manage_foreign_assets_for_local_consensus_p
+			Runtime,
+			AllPalletsWithoutSystem,
+			XcmConfig,
+			LocationToAccountId,
+			ForeignAssetsInstance,
+		>(
+			collator_session_keys().add(collator_session_key(BLOCK_AUTHOR_ACCOUNT)),
+			ExistentialDeposit::get(),
+			AccountId::from([73; 32]),
+			AccountId::from(BLOCK_AUTHOR_ACCOUNT),
+			// receiving ROCs
+			(MultiLocation { parents: 2, interior: X1(GlobalConsensus(Kusama)) }, 1000000000000, 1_000_000_000),
+			bridging_to_asset_hub_kusama,
+			(
+				X1(PalletInstance(bp_bridge_hub_polkadot::WITH_BRIDGE_POLKADOT_TO_KUSAMA_MESSAGES_PALLET_INDEX)),
+				GlobalConsensus(Kusama),
+				X1(Parachain(1000))
+			)
+		)
+}
+#[test]
+fn report_bridge_status_from_xcm_bridge_router_for_kusama_works() {
+	missing_asset_test_utils_test_cases_over_bridge::report_bridge_status_from_xcm_bridge_router_works::<
+		Runtime,
+		AllPalletsWithoutSystem,
+		XcmConfig,
+		LocationToAccountId,
+		ToKusamaXcmRouterInstance,
+	>(
+		collator_session_keys(),
+		bridging_to_asset_hub_kusama,
+		|| {
+			Decode::decode(&mut &bp_asset_hub_polkadot::CongestedMessage::get().encode()[..])
+				.unwrap()
+		},
+		|| {
+			Decode::decode(&mut &bp_asset_hub_polkadot::UncongestedMessage::get().encode()[..])
+				.unwrap()
+		},
+	)
+}
+
+#[test]
+fn test_report_bridge_status_call_compatibility() {
+	// if this test fails, make sure `bp_asset_hub_kusama` has valid encoding
+	assert_eq!(
+		RuntimeCall::ToKusamaXcmRouter(pallet_xcm_bridge_hub_router::Call::report_bridge_status {
+			bridge_id: Default::default(),
+			is_congested: true,
+		})
+		.encode(),
+		bp_asset_hub_polkadot::Call::ToKusamaXcmRouter(
+			bp_asset_hub_polkadot::XcmBridgeHubRouterCall::report_bridge_status {
+				bridge_id: Default::default(),
+				is_congested: true,
+			}
+		)
+		.encode()
+	)
+}
+
+#[test]
+fn check_sane_weight_report_bridge_status() {
+	use pallet_xcm_bridge_hub_router::WeightInfo;
+	let actual = <Runtime as pallet_xcm_bridge_hub_router::Config<
+			ToKusamaXcmRouterInstance,
+		>>::WeightInfo::report_bridge_status();
+	let max_weight = bp_asset_hub_polkadot::XcmBridgeHubRouterTransactCallMaxWeight::get();
+	assert!(
+		actual.all_lte(max_weight),
+		"max_weight: {:?} should be adjusted to actual {:?}",
+		max_weight,
+		actual
+	);
+}
+
+#[test]
+fn change_xcm_bridge_hub_router_byte_fee_by_governance_works() {
+	asset_test_utils::test_cases::change_storage_constant_by_governance_works::<
+		Runtime,
+		bridging::XcmBridgeHubRouterByteFee,
+		Balance,
+	>(
+		collator_session_keys(),
+		1000,
+		Box::new(|call| RuntimeCall::System(call).encode()),
+		|| {
+			(
+				bridging::XcmBridgeHubRouterByteFee::key().to_vec(),
+				bridging::XcmBridgeHubRouterByteFee::get(),
+			)
+		},
+		|old_value| {
+			if let Some(new_value) = old_value.checked_add(1) {
+				new_value
+			} else {
+				old_value.checked_sub(1).unwrap()
+			}
+		},
+	)
+}
+
+// missing stuff from asset_test_utils::test_cases_over_bridge
```

Comment:
nit: maybe add TODO here to remove with/when required deps update

Reply:
Done